src/ops/dbcsr_io.F

```diff
@@ -846,19 +848,19 @@ SUBROUTINE dbcsr_binary_read(filepath, distribution, groupid, matrix_new)
       row_blk_size => ginfo_vec(1:nblkrows_total)
       col_blk_size => ginfo_vec(nblkrows_total + 1:globalinfo_size)
 
-! compute the offsets where sub-header2 and sub-header3 start
-      subh2_start = (4 + globalinfo_size)*mpi_integer_size + char_count*mpi_character_size
-      subh3_start = subh2_start + 2*size_of_pgrid*mpi_integer_size
+      ! compute the offsets where sub-header2 and sub-header3 start
+      subh2_start = INT((4 + globalinfo_size)*mpi_integer_size + char_count*mpi_character_size, KIND=file_offset)
+      subh3_start = subh2_start + INT(2*size_of_pgrid*mpi_integer_size, KIND=file_offset)
 
-! compute the offsets in sub-header2 and read 2 integers nblocks, data_area_size
+      ! compute the offsets in sub-header2 and read 2 integers nblocks, data_area_size
       ! number of data chunks from sub-header 2 and 3 to be read by every node rounded up
       ! to the next integer to make it even for all the nodes in the specified mpi group
-      share_size = CEILING(REAL(size_of_pgrid, KIND=dp)/group_list_size)
+      share_size = CEILING(REAL(size_of_pgrid, KIND=dp)/REAL(group_list_size, KIND=dp))
 
       ALLOCATE (subh2_offsets(share_size))
       subh2_offsets = BOF
       DO i = 1, share_size
-         offset = subh2_start + mpi_integer_size*2*(worker_id + (i - 1)*group_list_size)
+         offset = subh2_start + mpi_integer_size*2*INT(worker_id + (i - 1)*group_list_size, KIND=file_offset)
```

Comment:
not sure what the compiler did with the original version here: first `worker_id + (i - 1)*group_list_size` and then converting to larger size, or converting everything to larger size.